homeassistant/components/fritz/switch.py

```diff
@@ -0,0 +1,644 @@
+                    FritzBoxDeflectionSwitch(
+                        fritzbox_tools, device_friendly_name, dict_of_deflection
+                    )
+                )
+
+    return entities_list
+
+
+def port_entities_list(
+    fritzbox_tools: FritzBoxTools, device_friendly_name: str
+) -> list[FritzBoxPortSwitch]:
+    """Get list of port forwarding entities."""
+
+    _LOGGER.debug("Setting up %s switches", SWITCH_TYPE_PORTFORWARD)
+    entities_list: list = []
+    service_name = "Layer3Forwarding"
+    connection_type = service_call_action(
+        fritzbox_tools, service_name, "1", "GetDefaultConnectionService"
+    )
+    if not connection_type:
+        _LOGGER.debug("The FRITZ!Box has no %s options", SWITCH_TYPE_PORTFORWARD)
+        return []
+
+    # Return NewDefaultConnectionService sample: "1.WANPPPConnection.1"
+    con_type: str = connection_type["NewDefaultConnectionService"][2:][:-2]
+
+    # Query port forwardings and setup a switch for each forward for the current device
+    resp = service_call_action(
+        fritzbox_tools, con_type, "1", "GetPortMappingNumberOfEntries"
+    )
+    if not resp:
+        _LOGGER.debug("The FRITZ!Box has no %s options", SWITCH_TYPE_DEFLECTION)
+        return []
+
+    port_forwards_count: int = resp["NewPortMappingNumberOfEntries"]
+
+    _LOGGER.debug(
+        "Specific %s response: GetPortMappingNumberOfEntries=%s",
+        SWITCH_TYPE_PORTFORWARD,
+        port_forwards_count,
+    )
+
+    for i in range(port_forwards_count):
+
+        portmap = service_call_action(
+            fritzbox_tools,
+            con_type,
+            "1",
+            "GetGenericPortMappingEntry",
+            NewPortMappingIndex=i,
+        )
+
+        if not portmap:
+            _LOGGER.debug("The FRITZ!Box has no %s options", SWITCH_TYPE_DEFLECTION)
+        else:
+            _LOGGER.debug(
+                "Specific %s response: GetGenericPortMappingEntry=%s",
+                SWITCH_TYPE_PORTFORWARD,
+                portmap,
+            )
+
+            # We can only handle port forwards of the given device
+            local_ip = async_get_source_ip(fritzbox_tools.host)
+            _LOGGER.debug("IP source for %s is %s", fritzbox_tools.host, local_ip)
+            if portmap["NewInternalClient"] == local_ip:
+                entities_list.append(
+                    FritzBoxPortSwitch(
+                        fritzbox_tools,
+                        device_friendly_name,
+                        portmap,
+                        i,
+                        con_type,
+                    )
+                )
+
+    return entities_list
+
+
+def profile_entities_list(
+    fritzbox_tools: FritzBoxTools, device_friendly_name: str
+) -> list[FritzBoxProfileSwitch]:
+    """Get list of profile entities."""
+    _LOGGER.debug("Setting up %s switches", SWITCH_TYPE_DEVICEPROFILE)
+    entities_list: list = []
+    if len(fritzbox_tools.fritz_profiles) > 0:
```

Comment:
Could reverse the condition and return early.. but doesn't save you much indent here

Reply:
Done ;-)